core/src/mindustry/logic/LExecutor.java

```diff
@@ -1910,6 +1911,36 @@ public void run(LExecutor exec){
         }
     }
 
+    public static class PlaySoundI implements LInstruction{
+        public boolean positional;
+        public LVar id, volume, pitch, pan, x, y;
+
+        public PlaySoundI(){
+        }
+
+        public PlaySoundI(boolean positional, LVar id, LVar volume, LVar pitch, LVar pan, LVar x, LVar y){
+            this.positional = positional;
+            this.id = id;
+            this.volume = volume;
+            this.pitch = pitch;
+            this.pan = pan;
+            this.x = x;
+            this.y = y;
+        }
+
+        @Override
+        public void run(LExecutor exec){
+            Sound sound = Sounds.getSound(id.numi());
+            if(sound == null || sound == Sounds.swish) sound = Sounds.none; //no.
+            
+            if(positional){
+                sound.at(World.unconv(x.numf()), World.unconv(y.numf()), pitch.numf(), volume.numf());
```

Comment:
Maybe it's worth limiting volume to something like 2-3 to prevent troll maps with ear-destroying volume.